homeassistant/components/tts/yandextts.py

```diff
@@ -0,0 +1,119 @@
+"""
+Support for the yandex speechkit tts  service.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/tts/yandextts/
+"""
+import asyncio
+import logging
+
+import aiohttp
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.const import CONF_API_KEY
+from homeassistant.components.tts import Provider, PLATFORM_SCHEMA, CONF_LANG
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+
+_LOGGER = logging.getLogger(__name__)
+
+YANDEX_API_URL = "https://tts.voicetech.yandex.net/generate?"
+
+SUPPORT_LANGUAGES = [
+    'ru-RU', 'en-US', 'tr-TR', 'uk-UK'
+]
+
+SUPPORT_CODECS = [
+    'mp3', 'wav', 'opus',
+]
+
+SUPPORT_VOICES = [
+    'jane', 'oksana', 'alyss', 'omazh',
+    'zahar', 'ermil'
+]
+CONF_CODEC = 'codec'
+CONF_VOICE = 'voice'
+
+DEFAULT_LANG = 'en-US'
+DEFAULT_CODEC = 'mp3'
+DEFAULT_VOICE = 'zahar'
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Optional(CONF_LANG, default=DEFAULT_LANG): vol.In(SUPPORT_LANGUAGES),
+    vol.Optional(CONF_CODEC, default=DEFAULT_CODEC): vol.In(SUPPORT_CODECS),
+    vol.Optional(CONF_VOICE, default=DEFAULT_VOICE): vol.In(SUPPORT_VOICES),
+})
+
+
+@asyncio.coroutine
+def async_get_engine(hass, config):
+    """Setup VoiceRSS speech component."""
+    return YandexSpeechKitProvider(hass, config)
+
+
+class YandexSpeechKitProvider(Provider):
+    """VoiceRSS speech api provider."""
+
+    def __init__(self, hass, conf):
+        """Init VoiceRSS TTS service."""
+        self.hass = hass
+        self._codec = conf.get(CONF_CODEC)
+        conf.get(CONF_API_KEY)
+        self._url = '{baseUrl}format={format}&speaker={speaker}&key={key}' \
```

Comment:
Look at tts/google.py I think it will be more clear to use param on websocket.request instead str.format